src/sourcerer/infrastructure/access_credentials/services.py

```
@@ -176,23 +177,26 @@ def authenticate(self, credentials: str):  # type: ignore
-        return Boto3Credentials(
-            session=session,
-            endpoint_url=credentials.get("endpoint_url", None),
-            signature_version=credentials.get("signature_version", None),
-        )
+            return Boto3Credentials(
+                session=session,
+                endpoint_url=credentials.get("endpoint_url", None),
+                signature_version=credentials.get("signature_version", None),
+            )
+        except Exception as e:
+            raise CredentialsAuthError("Failed to authenticate") from e
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)